matchers/php/todomatcher.go

```diff
@@ -2,37 +2,83 @@ package php
 
 import (
 	"regexp"
+	"sync"
 
 	"github.com/preslavmihaylov/todocheck/matchers/errors"
 )
 
-var singleLineTodoPattern = regexp.MustCompile("^\\s*//.*TODO")
-var singleLineValidTodoPattern = regexp.MustCompile("^\\s*// TODO (#?[a-zA-Z0-9\\-]+):.*")
+var lock sync.Mutex
 
-var singleLineScriptTodoPattern = regexp.MustCompile("^\\s*#.*TODO")
-var singleLineScriptValidTodoPattern = regexp.MustCompile("^\\s*# TODO (#?[a-zA-Z0-9\\-]+):.*")
+var singleLineTodoPattern *regexp.Regexp
+var singleLineValidTodoPattern *regexp.Regexp
 
-var multiLineTodoPattern = regexp.MustCompile("(?s)^\\s*/\\*.*TODO")
-var multiLineValidTodoPattern = regexp.MustCompile("(?s)^\\s*/\\*.*TODO (#?[a-zA-Z0-9\\-]+):.*")
+var singleLineScriptTodoPattern *regexp.Regexp
+var singleLineScriptValidTodoPattern *regexp.Regexp
+
+var multiLineTodoPattern *regexp.Regexp
+var multiLineValidTodoPattern *regexp.Regexp
 
 // NewTodoMatcher for php comments
-func NewTodoMatcher() *TodoMatcher { return &TodoMatcher{} }
+func NewTodoMatcher(todos string) *TodoMatcher {
+	lock.Lock()
+	defer lock.Unlock()
+
+	// Single line
+	if singleLineTodoPattern == nil {
+		singleLineTodoPattern = regexp.MustCompile("^\\s*//.*" + todos)
+	}
+	if singleLineValidTodoPattern == nil {
+		singleLineValidTodoPattern = regexp.MustCompile("^\\s*// " + todos + " (#?[a-zA-Z0-9\\-]+):.*")
+	}
+
+	// Script line
+	if singleLineScriptTodoPattern == nil {
+		singleLineScriptTodoPattern = regexp.MustCompile("^\\s*#.*" + todos)
+	}
+	if singleLineScriptValidTodoPattern == nil {
+		singleLineScriptValidTodoPattern = regexp.MustCompile("^\\s*# " + todos + " (#?[a-zA-Z0-9\\-]+):.*")
+	}
+
+	// Multiline line
+	if multiLineTodoPattern == nil {
+		multiLineTodoPattern = regexp.MustCompile("(?s)^\\s*/\\*.*" + todos)
+	}
+	if multiLineValidTodoPattern == nil {
+		multiLineValidTodoPattern = regexp.MustCompile("(?s)^\\s*/\\*.*" + todos + " (#?[a-zA-Z0-9\\-]+):.*")
+	}
```

Comment:
Once we remove the global variables, all these can be condensed into a single return statement, which creates the struct.